File: src/main/java/com/github/njuro/jard/base/BaseRepository.java
package com.github.njuro.jard.base;

import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.NoRepositoryBean;

/** Base repository for manipulating with JPA entites inheriting from {@link BaseEntity}. */
@NoRepositoryBean
public interface BaseRepository<ENTITY extends BaseEntity> extends JpaRepository<ENTITY, UUID> {}

File: src/main/kotlin/com/github/njuro/jard/rewrite/base/BaseRepository.kt
package com.github.njuro.jard.rewrite.base

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.NoRepositoryBean
import java.util.UUID

/** Base repository for manipulating with JPA entities inheriting from [BaseEntity].  */
@NoRepositoryBean
interface BaseRepository<E : BaseEntity> : JpaRepository<E, UUID>
